app/src/main/java/mozilla/lockbox/store/ClipboardStore.kt

```diff
@@ -51,16 +51,19 @@ open class ClipboardStore(
     private fun addToClipboard(label: String, str: String) {
         val clip = ClipData.newPlainText(label, str)
         clipboardManager.primaryClip = clip
-        clearClipboard(clipDataClearDelay, str)
+        timedReplaceDirty("", str)
     }
 
-    private fun clearClipboard(delay: Long, clearStr: String) {
+    private fun timedReplaceDirty(dirty: String, clean: String, delay: Long = clipDataClearDelay){
         Handler().postDelayed({
-            val clipData = clipboardManager.primaryClip.getItemAt(0)
-            if (clipData.text == clearStr) {
-
-                clipboardManager.primaryClip = ClipData.newPlainText("", "")
-            }
+            replaceDirty(dirty, clean)
         }, delay)
     }
+
+    fun replaceDirty(dirty: String, clean: String) {
+        val clipData = clipboardManager.primaryClip.getItemAt(0)
+        if (clipData.text == clean) {
+            clipboardManager.primaryClip = ClipData.newPlainText("", dirty)
+        }
```

Comment:
The idea was to check if the current clipboard entry is the same as the one you try to remove it, and if yes then replace with a dirty one in this case "". This prevent that if in meantime the user added something else (related or not with this application) in the clipboard it will not be deleted

so i test that current clipboard entry is the same as the one added (username/password) and if yes only then update clipboard with dirty

Reply:
A couple of comments:

1) This phrasing is very confusing.  Most people expect to be checking for `dirty` and changing it to `clean`.  This function does the exact opposite, which is going to lead to bugs in our future.
2) I don't understand why there needs to be two parameters to this function; shouldn't the reset clip value always be ""?

Reply:
1. Agreed. 
2. The clean string was a nod to a possible future where the clipboard's initial value is _restored_ rather than wiped. The `clean` string now defaults to an empty string.